src/Core/mcts/MCTSInterface.cpp

```diff
@@ -0,0 +1,32 @@
+#include <MCTSInterface.h>
+#include <memory>
+
+MCTSInterface::MCTSInterface(EnvironmentBase &environment) : m_environment(environment){};
+
+State MCTSInterface::run(int n_searches) {
+    State initialState = m_environment.GetStartState();
+    std::vector<State> unvisited_child_states = m_environment.GetValidChildStates(initialState);
+    m_root = SearchNode::create_SearchNode(nullptr, initialState, false);
+    m_root->set_unvisited_child_states(unvisited_child_states);
+
+    while (!m_environment.GetValidChildStates(m_root->state).empty()) {
+        auto best_child = m_search(n_searches);
+        m_root = best_child;
+    }
+
+    return m_root->state;
+}
+
+std::shared_ptr<SearchNode> MCTSInterface::m_search(int n_searches) {
+
+    for (int i = 0; i < n_searches; i++) {
+        // TreePolicy runs to find an unexpanded node to expand
+        auto expandedNode = m_tree_policy(m_root);
```

Comment:
can't figure out where the tree policy is defined

Reply:
Virtual member functions defined in the header and implemented in deriving classes